src/components/select/AsyncSelect.js

```diff
@@ -18,7 +18,7 @@ class AsyncSelect extends PureComponent {
   }
 
   componentDidUpdate(prevProps) {
-    if (this.props.pageSize !== prevProps.pageSize) {
+    if (this.props.pageSize !== prevProps.pageSize || this.props.loadOptions !== prevProps.loadOptions) {
```

Comment:
I think this is the right way to do it, we need to reset the pagination as well.

(can you confirm @driesd)